cmake/GenerateVersion.cmake

```diff
@@ -33,7 +41,7 @@ endif()
 
 if(NOT "${NVIM_VERSION_HASH}" STREQUAL "${CURRENT_VERSION_HASH}")
   message(STATUS "Using NVIM_VERSION: ${NVIM_VERSION}")
-  file(WRITE "${OUTPUT}" "${NVIM_VERSION_STRING}")
+  file(APPEND "${OUTPUT}" "${NVIM_VERSION_STRING}")
```

Comment:
Basically, the `${OUTPUT}` file contains the "v0.10.0-dev-459+gcb9390868" string. If the SHA1 of this file changes, then we know that the string inside must've changed, and we therefore need to reconfigure/rebuild the relevant code containing the version.  This will now continually change the file, and will always rebuild.

Reply:
Got bogged down with a teams chat. 

> Basically, the `${OUTPUT}` file contains the "v0.10.0-dev-459+gcb9390868" string. If the SHA1 of this file changes, then we know that the string inside must've changed, and we therefore need to reconfigure/rebuild the relevant code containing the version. This will now continually change the file, and will always rebuild.

I hadn't considered that, I've pushed an update to hopefully resolve this.